keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1073 @@
+
+Pod status & nodeName updates are handled by `status` and `binding` subresource requests
+respectively, and are not checked against policies.
+
+Update requests to Pods and [PodTemplate resources](#podtemplate-resources) will reevaluate the full object
+against audit & warn policies, independent of which fields are being modified.
+
+#### Ephemeral Containers
+
+In the initial implementation, ephemeral containers will be subject to the same policy restrictions,
+and adding or updating ephemeral containers will require a full policy check.
+
+<<[UNRESOLVED]>>
+
+_Non-blocking for alpha. This should be resolved for beta._
+
+Once ephemeral containers allow [custom security contexts], it may be desirable to run an ephemeral
+container with higher privileges for debugging purposes. For example, CAP_SYS_PTRACE is forbidden by
+the baseline policy but can be useful in debugging. We could introduce yet-another-mode-label that
+only applies enforcement to ephemeral containers (defaults to the enforce policy).
+
+[custom security contexts]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/277-ephemeral-containers#configurable-security-policy
+
+One way this could be handled under the current model is:
+1. Exempt a special username (not one that can be authenticated directly) from policy enforcement,
+   e.g. `ops:privileged-debugger`
+2. Grant the special user permission to ONLY operate on the ephemeral containers subresource (it is
+   critical that they cannot create or update pods directly).
+3. Grant (real) users that should have privileged debug capability the ability to impersonate the
+   exempt user.
+
+We could consider ways to streamline the user experience of this, for instance adding a special RBAC
+binding that exempts users when operating on the ephemeral containers subresource (e.g. an
+`escalate-privilege` verb on the ephemeral containers subresource).
+
+<<[/UNRESOLVED]>>
+
+#### Other Pod Subresources
+
+Aside from ephemeral containers, the policy is not checked for any other Pod subresources (status,
+bind, logs, exec, attach, port-forward).
+
+Although annotations can be updated through the status subresource, the apparmor annotations are
+immutable and the seccomp annotations are deprecated and slated for removal in v1.23.
+
+### Pod Security Standards
+
+Policy level definitions are hardcoded and unconfigurable out of the box. However, the [Pod Security
+Standards] leave open ended guidance on a few items, so we must make a static decision on how to
+handle these elements:
+
+**HostPorts** - (baseline) HostPorts will be forbidden. This is a more niche feature, and violates
+the container-host boundary.
+
+**AppArmor** - (baseline) Allow anything except `unconfined`. Custom AppArmor profiles must be
+installed by the cluster admin, and AppArmor fails closed when a profile is not present.
+
+**SELinux** - (baseline) type may only be set to allowlisted values, level may be anything, user &
```

Comment:
I thought we agreed these selinux options were for both baseline and restricted.

Reply:
> `restricted` by definition includes all `baseline` restrictions, and may add additional restrictions. can make that explicit here if that's helpful

Yes please.  I didn't understand it that way as I read.